prifi-lib/client/client.go

```diff
@@ -209,7 +208,35 @@ func (p *PriFiLibClientInstance) ProcessDownStreamData(msg net.REL_CLI_DOWNSTREA
 
 		//pass the data to the VPN/SOCKS5 proxy, if enabled
 		if p.clientState.DataOutputEnabled {
-			p.clientState.DataFromDCNet <- msg.Data
+			if p.clientState.DisruptionProtectionEnabled {
+				if len(msg.Data) < 32 {
+					// The realy did not send the hash or atleast all of it
+					// TODO: How should I treat this
+					log.Error("The realy did not send the hash back.")
+					p.clientState.B_echo_last = 1
+				} else {
+					// Getting hash sent by realy
+					var data []byte
+					data = msg.Data
+					hash := data[:32]
+
+					// Getting previously calculated hash
+					previousHash := p.clientState.HASHFromPreviousMessage[:]
```

Comment:
Could you refactor to `HashFromPreviousMessage` ? There's no reason for the uppercase